src/dmd/expressionsem.d

```diff
@@ -5817,7 +5817,8 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             {
                 if (op.hasSideEffect)
                 {
-                    auto tmp = copyToTemp(0, "__assertOp", op);
+                    const stc = STC.scope_ | STC.exptemp | (op.isLvalue() ? STC.ref_ : STC.rvalue);
```

Comment:
I suspect it's the `scope_` here that causes the new object to be destroyed prematurely. I think this will happen on every platform, but Win32 is the only platform that has different calling conventions for extern(D) and extern(C).

Reply:
Good to know, I didnt understand how the calling conventions would interfere here.
I removed the `STC.scope_` (as it was a remnant of batteling `DIP1000`) and it seems to pass again.